js/ui/windowManager.js

```diff
@@ -609,6 +662,20 @@ const WindowManager = new Lang.Class({
             this._unminimizeWindowDone(shellwm, actor);
             this._mapWindowDone(shellwm, actor);
             this._destroyWindowDone(shellwm, actor);
+            this._rotateInCompleted(actor);
+            this._rotateOutCompleted(actor);
+
+            let activeWaitForRotateTimeout = this._rotateInTimeouts.filter(function(timeout) {
+                return timeout == actor._waitingRotateTimeout;
+            });
+
+            if (activeWaitForRotateTimeout.length === 1) {
+                GLib.source_remove(activeWaitForRotateTimeout[0]);
+                this._rotateInTimeouts = this.rotateInTimeouts.filter(function(timeout) {
```

Comment:
I think this could be made simpler by just using `indexOf()` and `splice()` to remove the element.